File: java/k00.interface/Rectangle.java
package shape_space;

public class Rectangle extends Shape {
   /******* CONSTRUCTORS *******/
   public Rectangle(Double length, Double width)  { super(length, width); }
   public Rectangle(Integer length, Integer width)  { super(length, width); }

   /******* METHODS *******/
   public Double calculateArea() {
      return (super.width * super.height);
   }
}

File: cs/k00.interface/Rectangle.cs
public class Rectangle : Shape {
  /******* CONSTRUCTORS *******/
  public Rectangle(double length, double width) : base(length, width) { /* nothing */ }
  public Rectangle(int length, int width) : base(length, width) { /* nothing */ }

  /******* METHODS *******/
  public override double calculateArea() {
    return (base.width * base.height);
  }
}
